### PR TITLE
Ria 3131 pre hearing to decision

### PR DESCRIPTION
**Pull request by bradsorour**
### JIRA link (if applicable) ###

[RIA-3131](https://tools.hmcts.net/jira/browse/RIA-3131)

### Change description ###

See Jira

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecisionAndReasonsGeneratedPreparer.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase = callback
+            .getCaseDetails()
+            .getCaseData();
+
+        if (asylumCase.read(CASE_FLAG_SET_ASIDE_REHEARD_EXISTS, YesOrNo.class).map(flag -> flag.equals(YesOrNo.YES)).orElse(true)) {
```
Comment (mkayad) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecisionAndReasonsGeneratedPreparer.java:46]: is this condition only for reheard case? if so "orElse(true)" should be "orElse(false)", isn't it?

Comment (bradsorour) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecisionAndReasonsGeneratedPreparer.java]: Yes you are correct. Well spotted, thanks.

**Review by mkayad — DISMISSED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecisionAndReasonsPreparer.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase = callback
+            .getCaseDetails()
+            .getCaseData();
+
+        if (asylumCase.read(CASE_FLAG_SET_ASIDE_REHEARD_EXISTS, YesOrNo.class).map(flag -> flag.equals(YesOrNo.YES)).orElse(true)) {
```
[the remark this answers is not in the document]
Comment (bradsorour) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DecisionAndReasonsPreparer.java]: The preparer needs to clear the page fields so they are blank in the case of FTPA reheard.

**Review by Srihari-Nakkala — DISMISSED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**